packages/native/src/useLinking.tsx

```diff
@@ -46,9 +46,16 @@ const createMemoryHistory = () => {
       const id = window.history.state?.id;
 
       if (id) {
-        const index = items.findIndex((item) => item.id === id);
+        let currentIndex = items.findIndex((item) => item.id === id);
+        if (currentIndex < 0) {
+          currentIndex = 0;
+        }
 
-        return index > -1 ? index : 0;
+        // Fix createMemoryHistory.index variable's value, because it will go out of sync, when navigating
+        // with the browser. The 'popstate' event not sets the createMemoryHistory.index variable, but
+        // calls this function, so the corrrect value can be set.
+        index = currentIndex;
```

Comment:
We shouldn't change the internal state inside a getter. Please move this logic to the `onPopState` function which is the appropriate place.

Reply:
Yes, that is the appropriate place. I fixed.